solidity-bindgen-macros/src/abi_gen.rs

```diff
@@ -0,0 +1,159 @@
+use crate::abi_json::{abi_from_json, Abi};
+use ethabi::param_type::{ParamType, Reader};
+use inflector::cases::snakecase::to_snake_case;
+use proc_macro2::{Ident, Span, TokenStream};
+use std::path::Path;
+
+pub fn abi_from_file(path: impl AsRef<Path>, span: Span) -> TokenStream {
+    let name = path
+        .as_ref()
+        .file_stem()
+        .unwrap()
+        .to_str()
+        .unwrap()
+        .to_owned();
+    let bytes = std::fs::read(path).unwrap();
+    let abis = abi_from_json(&bytes);
```

Comment:
Calling this `abis` confused me initially, because I'm used to talking about "ABI" when referring to the entire contract ABI. Perhaps call it `fn_abis` or something to make clear this is not multiple ABI files but entries for each function? `abi_entries` would be enother option.